chain/store/store.go

```diff
@@ -649,6 +650,12 @@ func (cs *ChainStore) takeHeaviestTipSet(ctx context.Context, ts *types.TipSet)
 		return nil
 	}
 
+	tskBlk, err := ts.Key().ToStorageBlock()
+	if err != nil {
+		log.Errorf("failed to create a block from tsk: %s", ts.Key())
+	}
+	_ = cs.chainLocalBlockstore.Put(ctx, tskBlk)
```

Comment:
We don't expect to have tipsets in the chainstore, they won't work with splitstore. I'd try to not rely on this, and use tipset keys in places where you need to refer to a tipsets - those weren't made to be referenced by cid

Reply:
I guess FEVM wants to refer to tipset keys by CIDs, which is similar to what you want here, see https://github.com/filecoin-project/lotus/pull/9687/files

You may want to talk to @Stebalien about more details

Reply:
Oh!! Actually we don't use this code at all. IPC initially targeted wallaby, and we may have left that behind while doing the rebase to master. I removed it from the PR. Thanks!